test/cases/enum.ts

```diff
@@ -20,10 +21,158 @@ export var schema = {
   "additionalProperties": false
 }
 
-export var types = `interface Enum {
+export var configurations = [
+  {
+    settings: {
+      useTypescriptEnums: false
+    },
+    types: `interface Enum {
   foo: "a" | "b" | "c";
   bar: number;
   baz: {
     a: number;
   };
 }`
+  }, 
+  {
+    settings: {
+      useTypescriptEnums: true
+    },
+    types: `enum Foo {
+  a,
+  b,
+  c
+}
+enum Bar {
+  One = 1,
+  Two = 2,
+  Three = 3
+}
+enum Baz {
+  [object Object],
+  [object Object],
+  [object Object]
+}
+interface Enum {
+  foo: Foo;
+  bar: Bar;
+  baz: Baz;
+}`
+  },
+    {
+    settings: {
+      useTypescriptEnums: true,
+      addEnumUtils: true
+    },
+    types: `enum Foo {
+  a,
+  b,
+  c
+}
+class FooUtil {
```

Comment:
This is another thing that's compelling in my particular use case.  I'm trying to be able to bridge Java and TS, and these parsing methods are something that would need to be maintained individually on each side of that bridge if not generated (since TS enums are integers and we lose the string values at runtime).  

I know that's not really your concern, but if you don't want it here I'll need to maintain my own fork that does this.